dist/svg.js

```diff
@@ -1541,19 +1542,22 @@ SVG.FX = SVG.invent({
 
     // set the internal animation pointer to the specified position and updates the visualisation
   , at: function(pos){
+      var durDivSpd = this.situation.duration/this.spd
+
       this.pos = pos
-      this.situation.start = +new Date - pos * this.situation.duration
-      this.situation.finish = this.situation.start + this.situation.duration
+      this.situation.start = +new Date - pos * durDivSpd
+      this.situation.finish = this.situation.start + durDivSpd
       return this.step(true)
     }
 
-    // speeds up the animation by the given factor
-    // this changes the duration of the animation
-  , speed: function(speed){
-      this.situation.duration = this.situation.duration * this.pos + (1-this.pos) * this.situation.duration / speed
-      this.situation.finish = this.situation.start + this.situation.duration
-      return this.at(this.pos)
+    // set/get the speed of the animation
+  , speed: function(spd){
```

Comment:
Are we really sure we want to allow that `speed(0)` will return the speed value? Thats not what I would expect because I passed (an invalid) parameter to the function. I guess we can map `speed(0)` to `stop()` or better `pause()` or just let js throw that division by zero error